sdk/python/kfp/v2/components/utils.py

```diff
@@ -70,6 +70,9 @@ def sanitize_task_name(name: str) -> str:
     """Sanitizes task name."""
     return maybe_rename_for_k8s(name)
 
+def sanitize_v1_name(name: str) -> str:
+    """Sanitizes name for v1."""
+    return name.lower().replace(' ','-')
```

Comment:
This might not be enough. IIUC, v1 yaml can even special characters in the name, which would result illegal identifier.

Reply:
Also, what if such a name appear in v2 yaml? do we throw an error?

Reply:
Used v1 check.

Reply:
Updated a new check to escape characters - since `maybe_rename_for_k8s` changes the underscore but we don't change that for title.